en/hardware/porting_guide_nuttx.md

```diff
@@ -0,0 +1,80 @@
+  * Build configuration: [boards/px4/fmu-v5/default.cmake](https://github.com/PX4/Firmware/blob/master/boards/px4/fmu-v5/default.cmake).
+
+## NuttX Menuconfig Setup
+
+To modify the NuttX OS configuration, you can use [menuconfig](https://bitbucket.org/nuttx/nuttx) using the PX4 shortcuts:
+```sh
+make px4_fmu-v5_default menuconfig
+make px4_fmu-v5_default qconfig
+```
+
+For fresh installs of PX4 onto Ubuntu using [ubuntu_sim_nuttx.sh](https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim_nuttx.sh) you will also need to install *kconfig* tools from [NuttX tools](https://bitbucket.org/nuttx/tools/src/master/).
+
+> **Note** The following steps are not required if using the [px4-dev-nuttx](https://hub.docker.com/r/px4io/px4-dev-nuttx/) docker container or have installed to macOS using our normal instructions (as these include`kconfig-mconf`). 
+
+Run the following commands from any directory:
+```sh
+git clone https://bitbucket.org/nuttx/tools.git
+cd tools/kconfig-frontends
+sudo apt install gperf
+./configure --enable-mconf --disable-nconf --disable-gconf --enable-qconf --prefix=/usr
+make
+sudo make install
+```
+
+The `--prefix=/usr` determines the specific installation location (which must be in the `PATH` environment variable).
+The `--enable-mconf` and `--enable-qconf` options will enable the `menuconfig` and `qconfig` options respectively.
+
+To run `qconfig` you may need to install additional Qt dependencies.
+
+### Bootloader
+
+First you will need a bootloader, which depends on the hardware target:
+- STM32H7: the bootloader is based on NuttX, and is included in the PX4 Firmware.
+  See [here](https://github.com/PX4/Firmware/tree/master/boards/holybro/durandal-v1/nuttx-config/bootloader) for an example.
+- For all other targets, https://github.com/PX4/Bootloader is used. See [here](https://github.com/PX4/Bootloader/pull/155/files) for an example how to add a new target.
+  Then checkout the [buiding and flashing instructions](../software_update/stm32_bootloader.md).
+
+### Firmware Porting Steps
+
+1. Make sure you have a working [development setup](../setup/dev_env.md) and installed the NuttX menuconfig tool (see above).
+1. Download the source code and make sure you can build an existing target:
+   ```bash
+   git clone --recursive https://github.com/PX4/Firmware.git
+   cd Firmware
+   make px4_fmu-v5
+   ```
+1. Find an existing target that uses the same (or a closely related) CPU type and copy it.
+   For example for STM32F7:
+   ```bash
+   mkdir boards/manufacturer
+   cp -r boards/px4/fmu-v5 boards/manufacturer/my-target-v1
+   ```
+   Change **manufacturer** to the manufacturer name and **my-target-v1** to your board name.
+   
+Next you need to go through all files under **boards/manufacturer/my-target-v1** and update them according to your board.
```

Comment:
The layout of this whole list was a little buggered. I THINK this is how you wanted it (two lists?). Can you look at the rendered view and check it is what you wanted?

Reply:
Yeah that is better, thanks.